pkg/plugins/golang/multi-module/v1alpha1/plugin.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	cfgv3 "sigs.k8s.io/kubebuilder/v3/pkg/config/v3"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/stage"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugins/golang"
+)
+
+const pluginName = "multi-module." + golang.DefaultNameQualifier
+
+var (
+	pluginVersion            = plugin.Version{Number: 1, Stage: stage.Alpha}
+	supportedProjectVersions = []config.Version{cfgv3.Version}
+	pluginKey                = plugin.KeyFor(Plugin{})
+)
+
+var _ plugin.Init = Plugin{}
+
+// Plugin implements the plugin.Full interface
+type Plugin struct {
+	initSubcommand
```

Comment:
I think we need to implement the edit as well so that users are able to edit a pre-existe project to enable it.

Reply:
also needed to implement webhook as it could be that the go.mod was outdated due to the webhook introducing more dependencies into the api folder. Btw I think this should really be something to think about in future plugins. I do not think that we should put the Webhooks into the API folder, rather something like a machinery or into the controller.

Reply:
> also needed to implement webhook as it could be that the go.mod was outdated due to the webhook introducing more dependencies into the api folder.

I think that makes sense. How this approach would work with webhooks?


> Btw I think this should really be something to think about in future plugins. I do not think that we should put the Webhooks into the API folder, rather something like a machinery or into the controller.

The webhooks are related to the apis ( their purpose is for example convert an api ) therefore they do not fit in the controller.

Reply:
Currently the API Modules just incorporate more dependencies than if they were purely containing the api. If we want to deviate from that I would have to override the entire api scaffolding, probably not something we want

Reply:
So, do you mean that if we want to use this plugin we cannot work with webhooks? 
is that? If so, we could clarify that in the docs and in have a subcommand createWebhook which would be called automatically and if so, we print a msg telling the users that and what needs to be done.

Reply:
Closing this one to try to sort it out.